elements-fun/src/confidential.rs

```diff
@@ -176,26 +120,15 @@ macro_rules! impl_confidential_commitment {
                             return Err(A::Error::custom("missing prefix"));
                         };
 
-                        match prefix {
-                            0 => Ok($name::Null),
-                            1 => {
-                                // Apply $explicit_fn to allow `Value` to swap the amount bytes
-                                match access.next_element()? {
-                                    Some(x) => Ok($name::Explicit($explicit_fn(x))),
-                                    None => Err(A::Error::custom("missing commitment")),
-                                }
-                            }
-                            p if p == $prefixA || p == $prefixB => match access.next_element()? {
-                                Some(y) => Ok($name::Confidential(p, y)),
-                                None => Err(A::Error::custom("missing commitment")),
-                            },
-                            p => {
-                                return Err(A::Error::custom(format!(
-                                    "invalid commitment, invalid prefix: 0x{:02x}",
-                                    p
-                                )))
-                            }
+                        if prefix != $prefixA && prefix != $prefixB {
+                            return Err(A::Error::custom("missing commitment"));
```

Comment:
:question: Do you mean "missing prefix"?

Reply:
Probably, this was copy pasted :D